source/gui/frmhelpgeneratorunit.pas

```diff
@@ -1574,7 +1574,8 @@ procedure TfrmHelpGenerator.MenuGenerateRunClick(Sender: TObject);
   { Switch to "Generate" page }
   lbNavigation.ItemIndex := lbNavigation.Items.IndexOfObject(pageGenerate);
   lbNavigationClick(nil);
-
+  Application.ProcessMessages;
+  Sleep(500);
   ButtonGenerateDocsClick(nil);
```

Comment:
You process messages (once), then hang for half of the second. Why is this necessary? 

If you have a case when this is needed, please add a comment describing it. Otherwise please revert  this change,

Reply:
Application.processmessages force to send the messages to the system and sleep let the the time to those messages to be treated and the to ui to be repaint correctly. Generating is huge and long process. Without that the ui is not refresh correctly, because messages are in and stay in queue until the generating process have finish.

Reply:
Thanks for explaining! I get the point of `Application.ProcessMessages`.

I still don't think `Sleep(500)` is useful. You wrote above """sleep let the the time to those messages to be treated and the to ui to be repaint correctly""" but that's not what the `Sleep` routine does. It merely hangs the process for half a second, nothing gets repainted during this time.

Can you test whether the `Sleep(500)` is really useful to help repaint? And on what platform?

Reply:
I removed the sleep, it seems not necessary, during my tests